test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/AbstractMethodIsolatedCloudIntegrationTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.integrationtests;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.kie.cloud.api.DeploymentScenarioBuilderFactory;
+import org.kie.cloud.api.scenario.DeploymentScenario;
+import org.kie.cloud.git.GitProvider;
+import org.kie.cloud.git.GitProviderService;
+import org.kie.cloud.integrationtests.util.ScenarioDeployer;
+
+public abstract class AbstractMethodIsolatedCloudIntegrationTest<T extends DeploymentScenario> extends AbstractCloudIntegrationTest {
```

Comment:
@sutaakar Looking at this PR after it's been merged just to understand the changes. Can you please explain the reasoning behind the name of `AbstractMethodIsolatedCloudIntegrationTest`. What's that isolated method?

Reply:
@jhrcek It means that every test method is executed in separated (isolated) scenario.